frontend/app/utils/route-utils.ts

```diff
@@ -100,3 +100,39 @@ export function getRouteByPath(pathname: string, routes: I18nRoute[]): I18nPageR
 function normalizePath(pathname: string): string {
   return pathname.replace(/\/+$/, '');
 }
+
+/**
+ * Extracts and returns the locale from the provided request URL.
+ *
+ * This function analyzes the pathname of the given `request` URL to determine the locale. It supports
+ * English (`'en'`) and French (`'fr'`). If the pathname does not start with `/en` or `/fr`, the function
+ * defaults to English (`'en'`).
+ *
+ * @param request - The HTTP request object containing the URL from which to extract the locale.
+ * @returns The detected locale, either `'en'` or `'fr'`. Defaults to `'en'` if no valid locale is found.
+ */
+export function getLocale(request: Request): 'en' | 'fr' {
```

Comment:
There's already a `getLanguage()` function in `app/utils/i18n-utils.ts`. Can you use that instead so we don't have multiple ways to get the same data?